### PR TITLE
TextBox and PasswordBox Placeholder text and header properties thru attached properties

### PR DESCRIPTION
**Pull request by fangeles02**
From: #30 

textbbox passwordbox

We can use the new functionalities by using the `TextBoxRevealStyle` (modified to handle placeholder and header) for TextBoxes and `PasswordBoxRevealStyle` (based on the `TextBoxRevealStyle`) and applying the attached properties.

`HeaderText`: Inserts label on top of the data entry control
`HeaderSize`: Specifies the size of the header text
`HeaderForegroundBrush`: Specifies the foreground brush of the header

`PlaceHolderText`: The control's placeholder text (watermark)
`PlaceHolderForegroundBrush`: Specifies the foreground brush of the placeholder

Sample usage:
```
<TextBox Name="TextBoxBox1" HorizontalAlignment="Left"  Margin="5"  Width="120" Style="{StaticResource TextBoxRevealStyle}" fw:TextBoxHelper.PlaceHolderText="Username" fw:TextBoxHelper.HeaderText="Enter username"/>
<PasswordBox Name="PasswordBox1" HorizontalAlignment="Left"  Margin="5"  Width="120" Style="{StaticResource PasswordBoxRevealStyle}" fw:TextBoxHelper.PlaceHolderText="Password" fw:TextBoxHelper.HeaderText="Enter password"/>
```

### CODE REVIEW

**Comment by sourcechord** [comment]
Thanks!!
I can't took time now. I'll review it in this week end :smile:
Please, wait for a while.
I had a quick glance throw, almost LGTM.

File: FluentWPF/Styles/TextBox.xaml [hunk 1]
```diff
@@ -60,4 +127,129 @@
...
+
+                    <Grid>
+                        <Grid.RowDefinitions>
+                            <RowDefinition Height="Auto"/>
+                            <RowDefinition/>
+                        </Grid.RowDefinitions>
+
+                        <TextBlock x:Name="Label_Header" Margin="2,0,0,5" Text="{TemplateBinding local:TextBoxHelper.HeaderText}" FontSize="{TemplateBinding local:TextBoxHelper.HeaderSize}" Foreground="{TemplateBinding local:TextBoxHelper.HeaderForegroundBrush}"/>
+
+                        <!--
```
Comment (sourcechord) [inline FluentWPF/Styles/TextBox.xaml]: Please remove unnessesery comment.

File: FluentWPF/Styles/TextBox.xaml [hunk 2]
```diff
@@ -44,7 +89,29 @@
...
+                            </Trigger.ExitActions>
                         </Trigger>
+
+
+                        <DataTrigger Binding="{Binding ElementName=Label_Header, Path=Text}" Value="">
+                            <Setter TargetName="Label_Header" Property="Visibility" Value="Collapsed"/>
+                        </DataTrigger>
+
+
+                        <DataTrigger Binding="{Binding Text.Count, RelativeSource={RelativeSource Self}}" Value="0">
```
Comment (sourcechord) [inline FluentWPF/Styles/TextBox.xaml]: It Works, but I couldn't explain this behavior.
Text doesn't have `Count` property. `Count` is method that is defined in IEnumerable<T>'s extension method.
IDE warns this code, too.
I think PasswordBox's code is more suitable for it.
```xml
<DataTrigger Binding="{Binding RelativeSource={RelativeSource Self}, Path=Text, Mode=OneWay}" Value="">
```

**Comment by sourcechord** [comment]
Hi, @fangeles02.
How about it?

If you change the above points, and commit&push to your `fangeles02:fluentTextBox` branch, I will merge it.
Thanks.

**Comment by fangeles02** [comment]
Hi SourceChord,

I'll try to commit and push tomorrow morning, I am on a long vacation that's why I haven't commit into it lately. ??


Thanks.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by sourcechord** [comment]
Thanks!
Don't worry, I'm not in a rush
Please commit when it is convenient for you.
I'm looking forward to it :smiley:

File: FluentWPF/Styles/TextBox.xaml (hunk 1 above)
Comment (fangeles02) [reply FluentWPF/Styles/TextBox.xaml]: Have applied changes on latest commit

**Comment by fangeles02** [comment]
Hi,

I have applied changes on latest commit 🙂

**Comment by sourcechord** [comment]
Hi,@fangeles02
Thanks for commit 😄 

CommandBar & AppBarButton is good idea.
But, it seems to be a work in progress, and your latest commit has too many features.

Pull Request should contains only one thing.([see also](http://blog.ploeh.dk/2015/01/15/10-tips-for-better-pull-requests/#9456245213574e5b879092ad3f658fc0 ))

* Style of TextBox/PasswordBox changes.(as I suggested)
* Add some controls.
  * CommandBar
  * AppBarButton
* PointarTracker's change

These changes should not be included same PullRequest.

Could you remove these changes?
Or, May I delete these points after merging?

Thanks.

**Comment by fangeles02** [comment]
Oops. Sorry about that, it was not intended to be included on commit, im still new to github and branches 😅 

I'll commit again tomorrow.
Thanks 😅

**Comment by fangeles02** [comment]
Hi,

I removed the AppBarButton and CommandBar on latest commit. (I think Visual Studio Team Explorer is more stable than github app lol)

Thanks.

**Comment by sourcechord** [comment]
Thanks 👍 
But, there is still unnessecery changes.

* Brushes.xaml
* RevealStyles.xaml.cs
* Button.xaml & PointerTracker.cs
  * These change seems that intended to support border color follows change of system theme.
  * But, currently RevealBrush doesn't follows system theme.
  * I'm now trying to fix RevealBrush's behavior.

Could you revert above changes?


I like Visual Studio's diff view, too 😄 
And, I also use [SourceTree](https://www.sourcetreeapp.com/).

By the way,we can look around whole changes of the PR by `file changes` tab.
https://github.com/sourcechord/FluentWPF/pull/31/files

It is very usefull for checking all changes of the pull request.

**Comment by fangeles02** [comment]
New Pull Requets
#35

**Pull request closed without merging**